src/nvim/syntax.c

```diff
@@ -257,6 +255,48 @@ static int current_flags = 0;
 static int current_seqnr = 0;
 static int current_sub_char = 0;
 
+static char *(cterm_color_names[28]) = {
+  "Black", "DarkBlue", "DarkGreen", "DarkCyan",
+  "DarkRed", "DarkMagenta", "Brown", "DarkYellow",
+  "Gray", "Grey",
+  "LightGray", "LightGrey", "DarkGray", "DarkGrey",
+  "Blue", "LightBlue", "Green", "LightGreen",
+  "Cyan", "LightCyan", "Red", "LightRed", "Magenta",
+  "LightMagenta", "Yellow", "LightYellow", "White", "NONE"
+};
+// terminals with less than 16 colors...
```

Comment:
@mhinz asked to be able to change cursor via setting ctermbg. ctermbg=ColorName is straightforward but `ctermbg=4` needed a function to convert the integer to the colorname.